File: src/main/java/com/eager/questioncloud/verification/domain/Email.java
package com.eager.questioncloud.verification.domain;

import lombok.Getter;

@Getter
public class Email {
    private String to;
    private String subject;
    private String content;

    public Email(String to, String subject, String content) {
        this.to = to;
        this.subject = subject;
        this.content = content;
    }
}

File: qc-user/src/main/kotlin/com/eager/questioncloud/verification/domain/Email.kt
package com.eager.questioncloud.verification.domain

class Email(
    val to: String,
    val subject: String,
    val content: String
) {
    companion object {
        fun of(emailVerification: EmailVerification): Email {
            val template = com.eager.questioncloud.verification.template.EmailVerificationTemplateCreator.getTemplate(
                emailVerification.emailVerificationType,
                emailVerification.token
            )

            return Email(
                emailVerification.email,
                template.title,
                template.content
            )
        }
    }
}
